spec/javascript/components/supplemental_application/sections/RentalAssistance.test.js

```diff
@@ -25,119 +31,216 @@ const rentalAssistance = {
+      expect(mockCloseCallback.mock.calls.length).toEqual(0)
+      expect(mockDeleteCallback.mock.calls.length).toEqual(0)
+      findByNameAndProps(wrapper, 'Button', { text: 'Save' }).simulate('click')
+      expect(mockSaveCallback.mock.calls.length).toEqual(1)
+      expect(mockCloseCallback.mock.calls.length).toEqual(0)
+      expect(mockDeleteCallback.mock.calls.length).toEqual(0)
+    })
+
+    test('should call the cancel callback when clicked', () => {
+      expect(mockSaveCallback.mock.calls.length).toEqual(0)
+      expect(mockCloseCallback.mock.calls.length).toEqual(0)
+      expect(mockDeleteCallback.mock.calls.length).toEqual(0)
+      findByNameAndProps(wrapper, 'Button', { text: 'Cancel' }).simulate('click')
+      expect(mockSaveCallback.mock.calls.length).toEqual(0)
+      expect(mockCloseCallback.mock.calls.length).toEqual(1)
+      expect(mockDeleteCallback.mock.calls.length).toEqual(0)
+    })
   })
 
-  test('should call the appropriate callbacks for actions on edit assistance panel', () => {
-    const context = cloneDeep(baseContext)
-    const mockSaveCallback = jest.fn()
-    const mockDeleteCallback = jest.fn()
-    context.handleSaveRentalAssistance = mockSaveCallback
-    context.handleDeleteRentalAssistance = mockDeleteCallback
-    context.application.rental_assistances = [rentalAssistance]
-    context.showNewRentalAssistancePanel = false
-
-    const wrapper = mount(rentalAssistanceInContext(context))
-
-    // Check the call of the save callback upon existing rental assistance panel save
-    wrapper.find('button.button-link').first().simulate('click')
-    wrapper.find('RentalAssistanceForm button.primary').simulate('click')
-    expect(mockSaveCallback.mock.calls.length).toEqual(1)
-    expect(mockSaveCallback.mock.calls[0][2]).toEqual('update')
-
-    // Check the call of the delete callback upon existing rental assistance panel delete
-    wrapper.find('RentalAssistanceForm button.alert-fill').simulate('click')
-    expect(mockDeleteCallback.mock.calls.length).toEqual(1)
+  describe('when it already exists on salesforce', () => {
+    let mockSaveCallback
+    let mockCloseCallback
+    let mockDeleteCallback
+    let wrapper
+
+    beforeEach(() => {
+      mockSaveCallback = jest.fn()
+      mockCloseCallback = jest.fn()
+      mockDeleteCallback = jest.fn()
+
+      wrapper = getWrapper({
+        assistance: rentalAssistance,
+        isNew: false,
+        onSave: mockSaveCallback,
+        onClose: mockCloseCallback,
+        onDelete: mockDeleteCallback
+      })
+    })
+
+    test('matches snapshot', () => {
+      expect(wrapper).toMatchSnapshot()
+    })
+
+    test('should render the correct modal ID', () => {
+      expect(wrapper.find('#rental-assistance-edit-form-0')).toHaveLength(1)
+    })
+
+    test('should render the delete button', () => {
+      expect(findByNameAndProps(wrapper, 'Button', { text: 'Delete' })).toHaveLength(1)
```

Comment:
I think findByNameAndProps makes this super readable, and now we don't have to put any selector ids in the code just for testing.

Reply:
I had to add the id selector for puppeteer e2e tests, not sure if this would work for puppeteer selecting but maybe it would

Reply:
@akegan I don't think it would unfortunately. I think we'll probably need those ids for puppeteer tests for now